gitlab/pkg/apis/sources/v1alpha1/gitlabsource_types.go

```diff
@@ -0,0 +1,181 @@
+type GitLabSourceSpec struct {
+	// ServiceAccountName holds the name of the Kubernetes service account
+	// as which the underlying K8s resources should be run. If unspecified
+	// this will default to the "default" service account for the namespace
+	// in which the GitLabSource exists.
+	// +optional
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+	// ProjectUrl is the url of the GitLab project for which we are interested
+	// to receive events from.
+	// Examples:
+	//   https://knative.dev/eventing-contrib/gitlab
+	// +kubebuilder:validation:MinLength=1
+	ProjectUrl string `json:"projectUrl"`
+
+	// EventType is the type of event to receive from Gitlab. These
+	// correspond to supported events to the add project hook
+	// https://docs.gitlab.com/ee/api/projects.html#add-project-hook
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:Enum=push_events,push_events_branch_filter,issues_events,confidential_issues_events,merge_requests_events,tag_push_events,note_events,job_events,pipeline_events,wiki_page_events
+	EventTypes []string `json:"eventTypes"`
+
+	// AccessToken is the Kubernetes secret containing the GitLab
+	// access token
+	AccessToken SecretValueFromSource `json:"accessToken"`
+
+	// SecretToken is the Kubernetes secret containing the GitLab
+	// secret token
+	SecretToken SecretValueFromSource `json:"secretToken"`
+
+	// SslVerify if true configure webhook so the ssl verification is done when triggering the hook
+	SslVerify bool `json:"sslverify,omitempty"`
+
+	// Sink is a reference to an object that will resolve to a domain
+	// name to use as the sink.
+	// +optional
+	Sink *duckv1beta1.Destination `json:"sink,omitempty"`
+}
+
+// SecretValueFromSource represents the source of a secret value
+type SecretValueFromSource struct {
+	// The Secret key to select from.
+	SecretKeyRef *corev1.SecretKeySelector `json:"secretKeyRef,omitempty"`
+}
+
+// GitLabSourceStatus defines the observed state of GitLabSource
+type GitLabSourceStatus struct {
+	// Conditions holds the state of a source at a point in time.
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	duckv1.Status `json:",inline"`
+
+	// ID of the project hook registered with GitLab
+	Id string `json:"Id,omitempty"`
+
+	// SinkURI is the current active sink URI that has been configured
+	// for the GitHubSource.
+	// +optional
+	SinkURI string `json:"sinkUri,omitempty"`
+}
+
+const (
+	// GitLabSourceConditionReady has status True when the
+	// GitLabSource is ready to send events.
+	GitLabSourceConditionReady = apis.ConditionReady
+
+	// GitLabSourceConditionSinkProvided has status True when the
+	// GitlabSource has been configured with a sink target.
+	GitLabSourceConditionSinkProvided apis.ConditionType = "SinkProvided"
+
+	// GitLabSourceConditionSecretProvided has status True when the
+	// GitlabSource can read secret with gitlab tokens.
+	GitLabSourceConditionSecretProvided apis.ConditionType = "SecretProvided"
+)
```

Comment:
While not strictly required, core sources usually keep these inside a file that follows the naming convention [`<type>_lifecycle.go`](https://github.com/knative/eventing/blob/master/pkg/apis/sources/v1alpha2/apiserver_lifecycle.go).